returnn/torch/engine.py

```diff
@@ -115,6 +127,31 @@ def train_epoch(self):
 
     print("Trained %i steps" % step_idx)
 
+    self._model_state = model.state_dict()
+    self._save_model_state_to_file(self._model_state)
+
+  def _load_model_state_from_file(self, epoch=None):
+    """
+    :param int epoch:
+    :return: state saved via previous self._save_model_state_to_file() call for given epoch
+    :rtype: collections.OrderedDict
+    """
+    filename = self.get_epoch_model_filename(epoch=epoch) + util.get_model_filename_postfix()
+    print("Load model %s" % (filename,), file=log.v4)
+    return torch.load(filename)
+
+  def _save_model_state_to_file(self, model_state_dict):
```

Comment:
I would keep this more consistent to the TF/Theano logic, and directly store it based on the current loaded model, not via model state dict.